src/openms/include/OpenMS/ANALYSIS/ID/BayesianProteinInferenceAlgorithm.h

```diff
@@ -0,0 +1,125 @@
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Julianus Pfeuffer $
+// $Authors: Julianus Pfeuffer $
+// --------------------------------------------------------------------------
+#ifndef OPENMS_ANALYSIS_ID_BAYESIANPROTEININFERENCE_H
+#define OPENMS_ANALYSIS_ID_BAYESIANPROTEININFERENCE_H
+
+//#define INFERENCE_BENCH
+
+#include <OpenMS/ANALYSIS/ID/MessagePasserFactory.h>
+#include <OpenMS/DATASTRUCTURES/DefaultParamHandler.h>
+#include <OpenMS/CONCEPT/ProgressLogger.h>
+#include <OpenMS/MATH/MISC/GridSearch.h>
+#include <vector>
+#include <functional>
+#include <boost/optional.hpp>
+
+namespace OpenMS
+{
+  class ConsensusMap;
+  class IDBoostGraph;
+  class PeptideIdentification;
+  class ProteinIdentification;
+  class ExperimentalDesign;
+
+  class OPENMS_DLLAPI BayesianProteinInferenceAlgorithm :
+      public DefaultParamHandler,
+      public ProgressLogger
+  {
+  public:
+    /// Constructor
+    explicit BayesianProteinInferenceAlgorithm(unsigned int debug_lvl = 0);
+
+    /// Destructor
+    ~BayesianProteinInferenceAlgorithm() override = default;
+
+    void updateMembers_() override;
+
+    //Note: How to perform group inference
+    // Three options:
+    // - (as implemented) use the automatically created indist. groups and report their posterior
+    // - (can be done additionally) collapse proteins to groups beforehand and run inference
+    // - (if no single protein scores wanted at all) calculate prior from proteins for the group
+    //  beforehand and remove proteins from network (saves computation
+    //  because messages are not passed from prots to groups anymore.
+
+    /// A function object to pass into the IDBoostGraph class to perform algorithms on
+    /// connected components
+    class GraphInferenceFunctor;
+
+    /// A function object to pass into the IDBoostGraph class to perform algorithms on
+    /// connected components
+    class ExtendedGraphInferenceFunctor;
+
+    /// A function object to pass into the GridSearch;
+    struct GridSearchEvaluator;
+
+    /// Perform inference. Writes its results into protein and (optionally) peptide hits (new score).
+    /// Optionally adds indistinguishable protein groups with separate scores, too.
+    /// Currently only takes first proteinID run.
+    /// TODO loop over all runs
```

Comment:
is an error / warning given of more than one run?

Reply:
Now it is.